单独使用 MFSU ， 报错  __webpack_require__.h is not a function

<!--
感谢您向我们反馈问题，为了高效的解决问题，我们期望你能提供以下信息：
-->

## What happens?
单独使用 @umijs/mfsu 4.0.34 , 编译没有问题，
但是打开页面报错了
```
const mfsu = new MFSU({
    implementor: webpack,
    buildDepWithESBuild: false,
    // runtimePublicPath: '/',
    // cwd: process.cwd()
})
```
这里 把buildDepWithESBuild 设置为 false , 就会报如下错误
设置为true 就没问题
我不得不设置为false, 请见谅，因为在真实项目中 esbuild编译dependences中有的 js 文件有引入 css\png\jpeg\gif\less等
导致 esbuild 编译报没有相关loader 处理这些文件

## Mini Showcase Repository(REQUIRED)

> Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) then upload to your GitHub. 请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example)，并上传到你的 GitHub 仓库

<!-- 为节约大家的时间，无复现步骤的 ISSUE 会被关闭，提供之后再 REOPEN -->
<!-- YOUR_REPOSITORY_URL on github or stackbliz -->
复现仓库：https://github.com/flycj/umi-mfsu-single-example/tree/master

## How To Reproduce
1、下载项目后， `yarn` 安装依赖
2、`yarn dev`

**Expected behavior** 1. 2.
希望 yarn dev 之后页面能够正常打开

<!-- 请提供复现链接/步骤，错误日志以及相关配置 -->
1、目前 yarn dev 之后打开页面报如下错误

```
index.js:43 Uncaught TypeError: __webpack_require__.h is not a function
    at ./node_modules/webpack-dev-server/client/index.js?protocol=ws%3A&hostname=0.0.0.0&port=8899&pathname=%2Fws&logging=info&overlay=true&reconnect=10&hot=true&live-reload=true (index.js:43:1)
    at __webpack_require__ (mf-va_remoteEntry.js:80:41)
    at __webpack_exec__ (mf-va_remoteEntry.js":14:1)
    at mf-va_remoteEntry.js":14:1
    at webpackJsonpCallback (mf-va_remoteEntry.js:362:39)
    at Array.forEach (<anonymous>)
    at mf-va_remoteEntry.js:376:31
    at mf-va_remoteEntry.js:378:12
    at mf-va_remoteEntry.js:388:12
./node_modules/webpack-dev-server/client/index.js?protocol=ws%3A&hostname=0.0.0.0&port=8899&pathname=%2Fws&logging=info&overlay=true&reconnect=10&hot=true&live-reload=true @ index.js:43
__webpack_require__ @ mf-va_remoteEntry.js:80
__webpack_exec__ @ mf-va_remoteEntry.js":14
（匿名） @ mf-va_remoteEntry.js":14
webpackJsonpCallback @ mf-va_remoteEntry.js:362
（匿名） @ mf-va_remoteEntry.js:376
（匿名） @ mf-va_remoteEntry.js:378
（匿名） @ mf-va_remoteEntry.js:388
mf-va_remoteEntry.js":1 Uncaught (in promise) ScriptExternalLoadError: Loading script failed.
(missing: http://localhost:8899/mf-va_remoteEntry.js)
while loading "./@babel/runtime-corejs3/helpers/asyncToGenerator" from webpack/container/reference/mf
while loading "./@babel/runtime-corejs3/regenerator" from webpack/container/reference/mf
while loading "./@babel/runtime-corejs3/core-js-stable/promise" from webpack/container/reference/mf
while loading "./react" from webpack/container/reference/mf
while loading "./react-dom/client" from webpack/container/reference/mf
while loading "./react-router-dom" from webpack/container/reference/mf
    at webpack/container/reference/mf (mf-va_remoteEntry.js":1:1)
    at __webpack_require__ (bootstrap:24:1)
    at handleFunction (remotes loading:63:1)
    at remotes loading:82:1
    at Array.forEach (<anonymous>)
    at __webpack_require__.f.remotes (remotes loading:45:1)
    at ensure chunk:6:1
    at Array.reduce (<anonymous>)
    at __webpack_require__.e (ensure chunk:5:1)
    at fn.e (hot module replacement:82:1)
webpack/container/reference/mf @ mf-va_remoteEntry.js":1
__webpack_require__ @ bootstrap:24
handleFunction @ remotes loading:63
（匿名） @ remotes loading:82
__webpack_require__.f.remotes @ remotes loading:45
（匿名） @ ensure chunk:6
__webpack_require__.e @ ensure chunk:5
fn.e @ hot module replacement:82
（匿名） @ app.js:1
__webpack_require__.a @ async module:49
./mfsu-virtual-entry/app.js @ app.da4406f92eccf9c602a9.js:10
__webpack_require__ @ bootstrap:24
__webpack_exec__ @ mf-va_remoteEntry.js":14
（匿名） @ mf-va_remoteEntry.js":14
webpackJsonpCallback @ jsonp chunk loading:562
（匿名） @ app.da4406f92eccf9c602a9.js:1
await in （匿名）（异步）
__webpack_require__.a @ async module:49
./mfsu-virtual-entry/app.js @ app.da4406f92eccf9c602a9.js:10
__webpack_require__ @ bootstrap:24
__webpack_exec__ @ mf-va_remoteEntry.js":14
（匿名） @ mf-va_remoteEntry.js":14
webpackJsonpCallback @ jsonp chunk loading:562
（匿名） @ app.da4406f92eccf9c602a9.js:1
```

## Context
@umijs/mfsu: 4.0.34
node: 16.15.1
platform: Mac os
webpack: 5.75.0
webpack-cli: 5.0.0

> @flycj 至少要等到 #10026 被 merge 后， 12 月 15 日发 `v4.0.37` 才能用。
> 
> 如果急用，先 patch-package 一下吧，这里有视频教程：https://pnpm.io/cli/patch ，Patch 包 `@umijs/bundler-esbuild` ，位置 `src/build.js` 里 esbuild 配置的部分。

我更新到了 v4.0.37， 然后如下配置
```
const mfsu = new MFSU({
    implementor: webpack,
    buildDepWithESBuild: true,
    depBuildConfig: {
        loader: {
            '.png': 'file',
            '.jpeg': 'file',
            '.gif': 'file',
            '.css': 'css',
        }
    }
    // runtimePublicPath: '/',
    // cwd: process.cwd()
})
```
依旧报如下错误
```
✘ [ERROR] [plugin style] Build failed with 2 errors:
node_modules/react-photoswipe/dist/photoswipe.css:1:3085: ERROR: No loader is configured for ".png" files: node_modules/react-photoswipe/dist/default-skin.png
node_modules/react-photoswipe/dist/photoswipe.css:1:7342: ERROR: No loader is configured for ".gif" files: node_modules/react-photoswipe/dist/preloader.gif

    style-stub:/Users/dzj/Desktop/tests/mfsu/node_modules/react-photoswipe/dist/photoswipe.css:3:28:
      3 │             import css from "/Users/dzj/Desktop/tests/mfsu/node_modules/react-photoswipe/dist/photoswipe.css"
        ╵                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  This error came from the "onLoad" callback registered here:

    node_modules/@umijs/bundler-esbuild/dist/plugins/style.js:90:6:
      90 │       onLoad({
         ╵       ~~~~~~

    at setup (/Users/dzj/Desktop/tests/mfsu/node_modules/@umijs/bundler-esbuild/dist/plugins/style.js:90:7)
    at handlePlugins (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1260:21)
    at buildOrServeImpl (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:951:5)
    at Object.buildOrServe (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:759:5)
    at /Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:2096:17
    at new Promise (<anonymous>)
    at Object.build (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:2095:14)
    at build (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1942:51)
    at build (/Users/dzj/Desktop/tests/mfsu/node_modules/@umijs/bundler-esbuild/dist/build.js:78:49)
    at async DepBuilder.buildWithESBuild (/Users/dzj/Desktop/tests/mfsu/node_modules/@umijs/mfsu/dist/depBuilder/depBuilder.js:72:5)
    at async DepBuilder.build (/Users/dzj/Desktop/tests/mfsu/node_modules/@umijs/mfsu/dist/depBuilder/depBuilder.js:138:9)
    at async MFSU.buildDeps (/Users/dzj/Desktop/tests/mfsu/node_modules/@umijs/mfsu/dist/mfsu/mfsu.js:196:7)

webpack compiled successfully
error - Error: Build failed with 1 error:
style-stub:/Users/dzj/Desktop/tests/mfsu/node_modules/react-photoswipe/dist/photoswipe.css:3:28: ERROR: [plugin: style] Build failed with 2 errors:
node_modules/react-photoswipe/dist/photoswipe.css:1:3085: ERROR: No loader is configured for ".png" files: node_modules/react-photoswipe/dist/default-skin.png
node_modules/react-photoswipe/dist/photoswipe.css:1:7342: ERROR: No loader is configured for ".gif" files: node_modules/react-photoswipe/dist/preloader.gif
    at failureErrorWithLog (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1575:15)
    at /Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1033:28
    at runOnEndCallbacks (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1447:61)
    at buildResponseToResult (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1031:7)
    at /Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:1143:14
    at responseCallbacks.<computed> (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:680:9)
    at handleIncomingPacket (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:735:9)
    at Socket.readFromStdout (/Users/dzj/Desktop/tests/mfsu/node_modules/esbuild/lib/main.js:656:7)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) {
  errors: [
    {
      detail: [Error],
      id: '',
      location: [Object],
      notes: [Array],
      pluginName: 'style',
      text: 'Build failed with 2 errors:\n' +
        'node_modules/react-photoswipe/dist/photoswipe.css:1:3085: ERROR: No loader is configured for ".png" files: node_modules/react-photoswipe/dist/default-skin.png\n' +
        'node_modules/react-photoswipe/dist/photoswipe.css:1:7342: ERROR: No loader is configured for ".gif" files: node_modules/react-photoswipe/dist/preloader.gif'
    }
  ],
  warnings: []
}
```
请问，要怎么配置呢？ 🙏，是我配置的不对还是什么呢？

--- a/packages/bundler-esbuild/src/plugins/style.ts
+++ b/packages/bundler-esbuild/src/plugins/style.ts
@@ -46,9 +46,20 @@ export function style({
         charset,
         minify,
         loader: {
+          // images
           '.svg': 'dataurl',
-          // file ?
+          '.png': 'dataurl',
+          '.jpg': 'dataurl',
+          '.jpeg': 'dataurl',
+          '.gif': 'dataurl',
+          '.ico': 'dataurl',
+          '.webp': 'dataurl',
+          // font
           '.ttf': 'dataurl',
+          '.otf': 'dataurl',
+          '.woff': 'dataurl',
+          '.woff2': 'dataurl',
+          '.eot': 'dataurl',
         },
       };
 

